docusaurus/docs/create-a-plugin/extend-a-plugin/add-logs-metrics-traces-for-backend-plugins.md

```diff
@@ -0,0 +1,421 @@
+
+### Implement metrics in your plugin
+
+[Grafana plugin SDK for Go](../../introduction/grafana-plugin-sdk-for-go.md) uses the [Prometheus instrumentation library for Go applications](https://github.com/prometheus/client_golang). Any custom metric registered with the [default registry](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#pkg-variables) will be picked up by the SDK and exposed through the [Collect metrics capability](../../introduction/backend.md#collect-metrics).
+
+For convenience, it's recommended to use the [promauto package](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus/promauto) when creating custom metrics since it automatically registers the metric in the [default registry](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#pkg-variables) and expose them to Grafana.
+
+**Example:**
+
+```go
+package plugin
+
+import (
+    "context"
+
+    "github.com/prometheus/client_golang/prometheus"
+    "github.com/prometheus/client_golang/prometheus/promauto"
+
+    "github.com/grafana/grafana-plugin-sdk-go/backend"
+)
+
+var queriesTotal = promauto.NewCounterVec(
+    prometheus.CounterOpts{
+        Namespace: "grafana_plugin",
+        Name:      "query",
+        Help:      "Total number of queries.",
+    },
+    []string{"query_type"},
+)
+
+func (ds *Datasource) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+    for _, q := range req.Queries {
+        queriesTotal.WithLabelValues(q.QueryType).Inc()
+    }
+}
+```
+
+### Best practices
+
+- Consider using the namespace `grafana_plugin` as that would prefix any defined metric names with `plugin`. This will make it clear for operators that any metric named `grafana_plugin` originates from a Grafana plugin.
+- Use snake case style when naming metrics, e.g. `http_request_duration_seconds` instead of `httpRequestDurationSeconds`.
+- Use snake case style when naming metric labels, e.g. `status_code` instead of `statusCode`.
+- If the metric type is a counter, name it with a `_total` suffix, e.g. `http_requests_total`.
+- If the metric type is a histogram and you're measuring duration, name it with a `_<unit>` suffix, e.g. `http_request_duration_seconds`.
+- If the metric type is a gauge, name it to denote it's a value that can increase and decrease , e.g. `http_request_in_flight`.
+
+#### Label values and high cardinality
+
+Be careful with what label values you add. Using/allowing too many label values could result in high cardinality problems.
+
+If label values originate from user input they should be validated and cleaned. Use `metricutil.SanitizeLabelName("<label value>")` from pkg/infra/metrics/metricutil package to sanitize label names. Very important to only allow a predefined set of labels to minimize the risk of high cardinality problems.
```

Comment:
```suggestion
If label values originate from user input they should be validated and cleaned. Use `metricutil.SanitizeLabelName("<label value>")` from pkg/infra/metrics/metricutil package to sanitize label names. It is very important to only allow a predefined set of labels to minimize the risk of high cardinality problems.
```

I also think it might be good to give an example issue resulting from high cardinality somewhere in this section. Could optionally link out to another resource such as https://docs.honeycomb.io/concepts/high-cardinality/